DirectProgramming/C++SYCL_FPGA/Tutorials/Features/optimization_levels/minimum_latency/src/CMakeLists.txt

```diff
@@ -0,0 +1,158 @@
+set(SOURCE_FILE minimum_latency.cpp)
+set(TARGET_NAME_NO_CONTROL no_control)
+set(TARGET_NAME_MINIMUM_LATENCY minimum_latency)
+set(TARGET_NAME_MANUAL_REVERT manual_revert)
+# Emulator target
+set(EMULATOR_TARGET_NO_CONTROL ${TARGET_NAME_NO_CONTROL}.fpga_emu)
+# Simulator targets
+set(SIMULATOR_TARGET_NO_CONTROL ${TARGET_NAME_NO_CONTROL}.fpga_sim)
+set(SIMULATOR_TARGET_MINIMUM_LATENCY ${TARGET_NAME_MINIMUM_LATENCY}.fpga_sim)
+set(SIMULATOR_TARGET_MANUAL_REVERT ${TARGET_NAME_MANUAL_REVERT}.fpga_sim)
+# FPGA Report targets
+set(FPGA_EARLY_IMAGE_NO_CONTROL ${TARGET_NAME_NO_CONTROL}_report.a)
+set(FPGA_EARLY_IMAGE_MINIMUM_LATENCY ${TARGET_NAME_MINIMUM_LATENCY}_report.a)
+set(FPGA_EARLY_IMAGE_MANUAL_REVERT ${TARGET_NAME_MANUAL_REVERT}_report.a)
+# FPGA Hardware targets
+set(FPGA_TARGET_NO_CONTROL ${TARGET_NAME_NO_CONTROL}.fpga)
+set(FPGA_TARGET_MINIMUM_LATENCY ${TARGET_NAME_MINIMUM_LATENCY}.fpga)
+set(FPGA_TARGET_MANUAL_REVERT ${TARGET_NAME_MANUAL_REVERT}.fpga)
+
+# FPGA board selection
+if(NOT DEFINED FPGA_DEVICE)
+    set(FPGA_DEVICE "intel_a10gx_pac:pac_a10")
+    set(DEVICE_FLAG "-DA10")
+    set(MANUAL_REVERT_FLAGS "-Xssfc-exit-fifo-type=default")
+    message(STATUS "FPGA_DEVICE was not specified.\
+                    \nConfiguring the design to run on the default FPGA board ${FPGA_DEVICE} (Intel(R) PAC with Intel Arria(R) 10 GX FPGA). \
+                    \nPlease refer to the README for information on board selection.")
+else()
+    if(FPGA_DEVICE MATCHES ".*a10*")
+        set(DEVICE_FLAG "-DA10")
+        set(MANUAL_REVERT_FLAGS "-Xssfc-exit-fifo-type=default")
+    elseif(FPGA_DEVICE MATCHES ".*s10*")
+        set(DEVICE_FLAG "-DS10")
+        set(MANUAL_REVERT_FLAGS "-Xshyper-optimized-handshaking=on -Xssfc-exit-fifo-type=default")
```

Comment:
Can hyper-optimized-handshaking be set on all targets?

Reply:
The flag can be set on all targets, but it only gets effective on S10 and Agilex.

Reply:
I think it makes sense to set it for all targets: we are to move to IPA soon, and IPA supports other devices than A10, S10 and Agilex, so I'm wondering how to handle these cases
So if could could set it to all targets, I think that would make our life easier

Reply:
Hum that's annoying leave it as it for now, I'll have to think about a long term solution, thanks!